src/main/java/com/autotune/analyzer/services/DSMetadataService.java

```diff
@@ -90,8 +91,25 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
 
             DataSourceInfo datasource = new ExperimentDBService().loadDataSourceFromDBByName(dataSourceName);
             if(null != datasource) {
-                new DataSourceManager().importMetadataFromDataSource(datasource);
                 DataSourceMetadataInfo dataSourceMetadata = new ExperimentDBService().loadMetadataFromDBByName(dataSourceName, "false");
+                if (null != dataSourceMetadata) {
```

Comment:
Move all of the code that handles the doPost to the relevant files under datasource